[Elixir] new concept exercise - `regular-expressions` (#1757)
exercism/v3: languages/elixir/exercises/concept/regular-expressions/.docs/hints.md

## General

- Review regular expression patterns from the introduction
  - Remember, when creating the pattern a string, you must escape some characters.
- Review the Elixir documentation:
  - [Regex][regex-docs]
  - [Sigils: Regular Expressions][sigils-regex]
- Check out these great regular expression resources:
  - [Regular-Expressions.info][website-regex-info]
  - [Rex Egg: The world's most tyrannosauical regex tutorial][website-rexegg]
  - [RegexOne: Learn Regular Expressions with simple, interactive exercises.][website-regexone]
  - [Regular Expressions 101: an online regex sandbox][website-regex-101]
  - [RegExr: an online regex sandbox][website-regexr]

> ## 1. Write patterns for each numeric component

Author: ```suggestion
## 1. Match the day, month, and year from a date
```

Reviewer: I like it!

> - remember to return a string representing the regular expression pattern
> - review how to create _character classes_ or use _shorthand character classes_
> - review _quantifiers_

Reviewer: ```suggestion
- Remember to return a string representing the regular expression pattern
- Review how to create _character classes_ or use _shorthand character classes_
- Review _quantifiers_
```

Author: ```suggestion
- Remember to return a string representing the regular expression pattern.
- Review how to create _character classes_ or use _shorthand character classes_.
- Review _quantifiers_.
```

> ## 2. Write patterns for the day and month names

Author: ```suggestion
## 2. Match the day of the week and the month of the year
```

> - review how to write a pattern to match _string literals_
> - review _alternations_

Reviewer: ```suggestion
- Review how to write a pattern to match _string literals_
- Review _alternations_
```

Author: ```suggestion
- Review how to write a pattern to match _string literals_.
- Review _alternations_.
```

> ## 3. Capture the numeric and named date components

Author: ```suggestion
## 3. Capture the day, month, and year
```

> - review how to write patterns for captures and named captures

Reviewer: ```suggestion
- Review how to write patterns for captures and named captures
```

Author: ```suggestion
- Review how to write patterns for captures and named captures.
```

> ## 4. Compose the parts to capture from the whole date

Author: ```suggestion
## 4. Combine the captures to capture the whole date
```

> - remember, string interpolation may be used to join strings

Reviewer: ```suggestion
- Remember, string interpolation may be used to join strings
```

Author: ```suggestion
- Remember, string interpolation may be used to join strings.
```

> ## 5. Match only the date

Author: ```suggestion
## 5. Narrow the capture to match only on the date
```

> - remembers, _anchors_ help to match the pattern to the **whole line**
> - string interpolation may be used in the regular expression sigil syntax

Reviewer: ```suggestion
- Remember, _anchors_ help to match the pattern to the **whole line**
- String interpolation may be used in the regular expression sigil syntax
```

Author: ```suggestion
- Remembers, _anchors_ help to match the pattern to the **whole line**.
- String interpolation may be used in the regular expression sigil syntax.
```
